### PR TITLE
Replaced Electrum with bitcoinlib

### LINKED ISSUE #3831: Explore libbitcoin as an alternative to Electrum
Currently, we use Electrum to provide the functionality to trade Bitcoin on our decentralized market. However, this library is suffering from an increasing number of issues when being integrated with Tribler. The main issue is that Electrum has moved to Python 3 and deprecated the 'old' Python 2.7 implementation, which will likely suffer from vulnerabilities which are not fixed. Besides that, the Electrum library adds many dependencies to the Tribler code base.

As an alternative, I found [libbitcoin](https://github.com/1200wd/bitcoinlib). A quick investigation shows that it supports the same features as Electrum (creating wallets, managing balance and fetching fee estimations). It also contains support for Litecoin/Dash wallets, supports Python 2.7 and has a nice code base. It is, however, a less popular alternative compared to Electrum.

Goal: investigate whether libbitcoin can replace Electrum

### PR DESCRIPTION
**Pull request by devos50**
In this PR, I replaced the Electrum library with bitcoinlib. Note that bitcoinlib is optional, however, we should ship it with the Tribler client on Windows and Mac. On Linux, it is optional and users can install the wallet themselves using pip (users will get a warning if the library is not installed when trying to create a Bitcoin wallet).

Another advantage of bitcoinlib is that it also supports Litecoin and Dash. Support for these wallets will be added in a later PR.

Fixes #3831

### CODE REVIEW

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

File: Tribler/Core/Modules/wallet/btc_wallet.py [hunk 1]
```diff
@@ -213,31 +64,10 @@ def get_balance(self):
...
-                server = self.get_daemon().get_server(config)
-                result = server.run_cmdline(options)
-
-                if not result[0]:  # Transaction failed
-                    return fail(RuntimeError(result[1]))
-
-                return succeed(str(result[1]))
+                self._logger.info("Creating Bitcoin payment with amount %f to address %s", amount, address)
+                tx = self.wallet.send_to(address, int(amount))
+                return str(tx.hash)
```
Comment (xoriole) [inline Tribler/Core/Modules/wallet/btc_wallet.py]: Should this be ` return succeed(str(tx.hash))`?

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
Ready for review. The uncovered lines are quite hard to cover with a test :/

**Pull request merged**